cleverhans/compat.py

```diff
@@ -81,6 +81,16 @@ def reduce_mean(input_tensor, axis=None, keepdims=None,
                          reduction_indices=reduction_indices)
 
 
+def reduce_prod(input_tensor, axis=None, keepdims=None,
+                name=None, reduction_indices=None):
+  """
+  Wrapper around the tf.reduce_prod to handle argument keep_dims
+  """
+  return reduce_function(tf.reduce_mean, input_tensor, axis=axis,
```

Comment:
Shouldn't this be ` return reduce_function(tf.reduce_prod, input_tensor, axis=axis,` ?

Reply:
Is the test sensitive enough if it did not break despite the function not being the right one?

Reply:
It seems that the random generated samples will not trigger adjusted stddev, so the test pass even if this is wrong. Maybe I can come up with some dedicated cases for this?